internal/status/cache.go

```diff
@@ -35,10 +35,11 @@ const ValidCondition ConditionType = "Valid"
 // NewCache creates a new Cache for holding status updates.
 func NewCache(gateway types.NamespacedName) Cache {
 	return Cache{
-		proxyUpdates: make(map[types.NamespacedName]*ProxyUpdate),
-		gatewayRef:   gateway,
-		routeUpdates: make(map[types.NamespacedName]*ConditionsUpdate),
-		entries:      make(map[string]map[types.NamespacedName]CacheEntry),
+		proxyUpdates:   make(map[types.NamespacedName]*ProxyUpdate),
+		gatewayRef:     gateway,
+		gatewayUpdates: make(map[types.NamespacedName]*GatewayConditionsUpdate),
```

Comment:
Shouldn't need this, I think we can use the ConditionsUpdate as-is.

Reply:
@stevesloka the existing `ConditionsUpdate()` takes a `RouteGatewayStatus` which differs from a `GatewayStatus` type. The new `GatewayConditionsAccessor()` is called by the GatewayAPIProcessor to calculate the gateway status conditions.

Reply:
I think the Route status and Gateway status are different enough that we need separate handlers for them (because Route status is effectively namespaced by the Gateway that the status is relevant to).

Reply:
The only difference is that gateway has a `listenerStatus`. We can separate, just trying to keep duplication to a minimum as a `GatewayStatus.Conditions` is exactly the same as routes.

Reply:
Maybe we can re-use the `Conditions` struct to eliminate the code duplication. My only worry is that we change something with a route status and that doesn't roll through to Gateway for the core bits that all use `meta/v1.Condition`.

Reply:
...but can change it up later if it comes up.

Reply:
The base Gateway status is actually pretty simple, but the Listener status is not. It's where all the Route stuff shows up, currently, so I think that having a way to distinguish between those is good.

Reply:
I think that we should get this working and come back for code duplication later.